src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -2279,7 +2290,7 @@ class VariantSampleMatrix(val hc: HailContext, val metadata: VSMMetadata,
       Array("v", "s"))
   }
 
-  def writeMetadata(dirname: String, nPartitions: Int) {
+  def writeMetadata(dirname: String, nPartitions: Int, partitionCounts: Array[Long]) {
```

Comment:
you can remove nPartitions as a parameter here, and just use partitionCounts.length below